src/wasm.h

```diff
@@ -858,7 +858,9 @@ class Memory : public Importable {
   static const Address::address_t kPageMask = ~(kPageSize - 1);
 
   struct Segment {
-    Expression* offset;
+    uint32_t flags = 0;
```

Comment:
how about a bool for each flag? (we don't need to mirror the binary format here). with that, can also just use `segment.hasMemInit` instead of calling a function.

Reply:
sounds good.